include/maidsafe/crux/socket.hpp

```diff
@@ -253,34 +254,32 @@ inline void socket::close() {
     keepalive_timer.stop();
     transmit_queue.shutdown();
 
+    auto m = multiplexer;
```

Comment:
I was meaning the part at the start of the function: https://github.com/inetic/MaidSafe-CRUX/blob/5c682b4b7f4dc9191d1e088262c486842540ced9/include/maidsafe/crux/socket.hpp#L251-L252

We're checking if the `multiplexer` is null at the start of close, but then using `pin` later.  Unless I'm missing something, couldn't `multiplexer` be valid when line 252 executes, but be null by the time we copy it at line 257?  I'd thought it might be safer to change this to:

    inline void socket::close() {
        auto pin = multiplexer;
        // Already closed?
        if (!pin) return;

I guess it's not an issue if `close()` is always executed on a strand?